src/main/scala/device/MemEncryptUtil.scala

```diff
@@ -0,0 +1,808 @@
+/***************************************************************************************
+* Copyright (c) 2024-2025 Institute of Information Engineering, Chinese Academy of Sciences
+*
+* XiangShan is licensed under Mulan PSL v2.
+* You can use this software according to the terms and conditions of the Mulan PSL v2.
+* You may obtain a copy of Mulan PSL v2 at:
+*          http://license.coscl.org.cn/MulanPSL2
+*
+* THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND,
+* EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT,
+* MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
+*
+* See the Mulan PSL v2 for more details.
+***************************************************************************************/
+
+package device
+
+import chisel3._
+import chisel3.util._
+import chisel3.util.HasBlackBoxResource
+import org.chipsalliance.cde.config.Parameters
+import freechips.rocketchip.amba.axi4._
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.util._
+import freechips.rocketchip.amba.apb._
+import freechips.rocketchip.tilelink.AXI4TLState
+
+class GetCKI extends Module {
+  val io = IO(new Bundle {
+    val countRoundIn = Input(UInt(5.W))
+    val ckiOut = Output(UInt(32.W))
+  })
+  val ckiOutReg= RegInit(0.U(32.W))
+  val ckiValuesVec = VecInit(Seq(
+    "h00070e15".U, "h1c232a31".U, "h383f464d".U, "h545b6269".U,
+    "h70777e85".U, "h8c939aa1".U, "ha8afb6bd".U, "hc4cbd2d9".U,
+    "he0e7eef5".U, "hfc030a11".U, "h181f262d".U, "h343b4249".U,
+    "h50575e65".U, "h6c737a81".U, "h888f969d".U, "ha4abb2b9".U,
+    "hc0c7ced5".U, "hdce3eaf1".U, "hf8ff060d".U, "h141b2229".U,
+    "h30373e45".U, "h4c535a61".U, "h686f767d".U, "h848b9299".U,
+    "ha0a7aeb5".U, "hbcc3cad1".U, "hd8dfe6ed".U, "hf4fb0209".U,
+    "h10171e25".U, "h2c333a41".U, "h484f565d".U, "h646b7279".U
+  ))
+  when(io.countRoundIn < 32.U) {
+    ckiOutReg := ckiValuesVec(io.countRoundIn)
+  } .otherwise {
+    ckiOutReg := 0.U
+  }
+  io.ckiOut := ckiOutReg
+}
+
+
+class SboxReplace extends Module {
```

Comment:
What is Sbox? We need some simplest comments here to help people to have a basic understanding of what work this code is doing.